src/main/java/edu/kis/powp/jobs2d/features/DriverChangeObserver.java

```diff
@@ -0,0 +1,29 @@
+package edu.kis.powp.jobs2d.features;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.drivers.DriverComposite;
+import edu.kis.powp.jobs2d.drivers.MacroDriver;
+import edu.kis.powp.jobs2d.drivers.adapter.LineDriverAdapter;
+import edu.kis.powp.observer.Subscriber;
+
+public class DriverChangeObserver implements Subscriber {
+
+    @Override
+    public void update() {
+        Job2dDriver currentDriver = DriverFeature.getDriverManager().getCurrentDriver();
+
+        if (currentDriver instanceof MacroDriver) {
```

Comment:
instanceof is a code smell for a bad OO design. Actually I do not see any reason for this Subscriber (DriverComposite with Macro feature could be simply added to drivers). 
If Macro could be added to any driver then Subscriber would be needed.